### PR TITLE
make Pixiu Admin config management finer-grained

### PR DESCRIPTION
**Pull request by ztelur**
### What this PR does:

- init from api_config fetched from etcd remotely
- wath api_config change event from etcd
- use gost etcd and zk dependency

### CODE REVIEW

File: pkg/config/api_config.go [hunk 1]
```diff
@@ -67,64 +80,116 @@ func LoadAPIConfigFromFile(path string) (*fc.APIConfig, error) {
...
 	}
+	// TODO: 其他需要从远端获取配置的初始化操作
 
+	go listenResourceAndMethodEvent(metaConfig.APIConfigPath)
+	// TODO: 其他监控配置的操作，比如 PluginGroup 等
 	return apiConfig, nil
 }
 
-func initAPIConfigFromString(content string) error {
+func initAPIConfigFromKVList(kList, vList []string) error {
```
[the remark this answers is not in the document]
Comment (ztelur) [reply pkg/config/api_config.go:103]: resource and method data store in etcd as parent path relation. with this design, it' a little difficulty to split this function when fetch data from etcd use prefix=true option

File: pkg/config/api_config.go [hunk 2]
```diff
@@ -67,64 +80,116 @@ func LoadAPIConfigFromFile(path string) (*fc.APIConfig, error) {
 
 // LoadAPIConfig load the api config from config center
 func LoadAPIConfig(metaConfig *model.APIMetaConfig) (*fc.APIConfig, error) {
-	client = etcdv3.NewConfigClient(
+	client, _ = etcdv3.NewConfigClientWithErr(
```
Comment (williamfeng323) [inline pkg/config/api_config.go:83]: should we handle the error return from the new client function?

Comment (williamfeng323) [reply pkg/config/api_config.go:83]: It also limited the config center can only be etcd. Could we get the config center client from something like getConfigCenterClient(some_indicator)? It is not generic enough I think.

File: pkg/config/api_config.go [hunk 3]
```diff
@@ -67,64 +80,116 @@ func LoadAPIConfigFromFile(path string) (*fc.APIConfig, error) {
...
+
+	for i, k := range kList {
+		v := vList[i]
 
-		valid := validateAPIConfig(apiConf)
-		if !valid {
-			return perrors.Errorf("api config not valid error %v", perrors.WithStack(err))
+		// handle resource
+		re := getCheckResourceRegexp()
+		if m := re.Match([]byte(k)); m {
```
Comment (williamfeng323) [inline pkg/config/api_config.go:114]: maybe we could use below judgment to reduce the complexity.
```
if m := re.Match([]byte(k)); m {
    continue
}
```

File: pkg/config/api_config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (williamfeng323) [reply pkg/config/api_config.go:103]: I agree with cityiron. the complexity of this function is high. If we introduce sonarqube, it cannot pass the check.

[the remark this answers is not in the document]
Comment (ztelur) [reply pkg/config/api_config.go:103]: ok, I will make handle resource and handle method two method

File: pkg/config/api_config.go (hunk 2 above)
Comment (ztelur) [reply pkg/config/api_config.go]: already handle the error return from the new client function

**Review by mark4z — APPROVED** [review]

File: pkg/registry/zookeeper.go [hunk 4]
```diff
@@ -17,14 +17,14 @@
 package registry
 
 import (
+	gxzookeeper "github.com/dubbogo/gost/database/kv/zk"
```
Comment (AlexStocks) [inline pkg/registry/zookeeper.go:20]: move it to the 2rd import block.

File: pkg/registry/zookeeper_test.go [hunk 5]
```diff
@@ -18,6 +18,7 @@ package registry
 
 import (
 	"fmt"
+	gxzookeeper "github.com/dubbogo/gost/database/kv/zk"
```
Comment (AlexStocks) [inline pkg/registry/zookeeper_test.go:21]: move it to 2rd import block

**Review by williamfeng323 — APPROVED** [review]

**Pull request merged**